Update various ZFS options...

sean- opened the pull request:
Notably unbreak the sync property on FreeBSD (and probably Illumous). A few other nits were made as well re: correcting/adding various ZFS options.

Commit:
Sync ZFS option handling, notably unbreak sync's options. sync=(on|off) doesn't work
for FreeBSD and probably doesn't for other ZFS variants. Accept on/off for now even though it's probably broken for the sake of preserving the API.

Commit:
Sync ZFS option handling, notably unbreak sync's options. sync=(on|off) doesn't work
for FreeBSD and probably doesn't for other ZFS variants. Accept on/off for now even though it's probably broken for the sake of preserving the API.

Commit:
Sync ZFS option handling, notably unbreak sync's options. sync=(on|off) doesn't work
for FreeBSD and probably doesn't for other ZFS variants. Accept on/off for now even though it's probably broken for the sake of preserving the API.

Commit:
I take that back, ZFS for Linux uses on/off.

Commit:
I take that back, ZFS for Linux uses on/off.

Commit:
I take that back, ZFS for Linux uses on/off.

christophebiocca committed:
Ensure named packages not in top level are marked as missing.

jimi-c committed:
Catch auth errors in winrm and properly raise errors

Fixes #7875

micw commented:
Meanwhile (at least with zfs 0.6.3) linux only accepts standard/default/disabled. So at the moment this option is broken. Please accept the pull request to get it fixed! Thank you!

jlaska committed:
Fix debian builds by correcting debian/changelog

Remove 1.8 debian/changelog entry with a date of "TBD" as this causes
the debian build process to fail.  We can add this back with the 1.8
release becomes a reality, or add it back now with a valid date in the
future.

jlaska committed:
Make debuild happy by including a date in debian/changelog

The previous commit resolved the debian build problem, but was building
packages with the stable version.  A development version is required in
the changelog to properly build development DEB packages.

Commit:
Add another ticket stub.

jimi-c committed:
Include *.ps1 files in the library for installation

Fixes #8490

jimi-c committed:
Merge pull request #8475 from christophebiocca/fix-npm-command

Ensure npm packages actually are installed in a reachable way.

jimi-c committed:
Make sure default vars are used in template calls for tasks

Fixes #8499

mpdehaan committed:
Update CHANGELOG.md

OH NO NAMING CONFLICT

mpdehaan committed:
Merge pull request #8488 from bekimb/docstring_basic_module

Keep docstring up to date with the code in basic module

mpdehaan committed:
Update no_thanks.md

Commit:
Merge branch 'devel' of github.com:sean-/ansible into devel

Commit:
Merge branch 'devel' of github.com:sean-/ansible into devel

sean- commented:
Merged, re-added 'on','off' as options to continue supporting Linux.

mpdehaan commented:
# Cleanup Request

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We do require submissions to be free of merge commits, which cause difficulties in reviewing and merging code, and also affect the history of the source tree.
To resolve this problem, it may be helpful to create a new branch and cherry-pick commits onto it.
- In general, always start a "topic branch" named after each thing you would like to contribute
- use "git pull --rebase" vs "git pull" to stay in tune with upstream, as well as "git rebase" vs "git merge"
- if you have things broken out over many commits, consider submitting just one squashed commit that keeps the history concise
- you can probably force push to this branch to fix up this pull request rather than submitting a new ticket

Just as a quick reminder of things, this is a really busy project.  We have over 800 contributors and to manage the queue effectively
we assign things a priority between P1 (highest) and P5.  We'd like to thank you very much for your time!
We'll work things in priority order, so just wanted you to be aware of the queue and know we haven't forgot about you!

We will definitely see your comments on this issue when reading this ticket, but may not be able to reply promptly.  You may also wish to join one of our two mailing lists
which is very active:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Thank you once again for this and your interest in Ansible!

mpdehaan commented:
Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

On September 26, 2014, due to enormous levels of contribution to the project Ansible decided to reorganize module repos, making it easier
for developers to work on the project and for us to more easily manage new contributions and tickets.

We split modules from the main project off into two repos, http://github.com/ansible/ansible-modules-core and http://github.com/ansible/ansible-modules-extras

If you still would like this pull request merged, we will need your help making this target the new repo.  If you do not take any action, this
pull request unfortunately cannot be applied.

We apologize that we are not able to make this transition happen seamlessly, though this is a one-time change and your help is greatly appreciated -- 
this will greatly improve velocity going forward.

Both sets of modules will ship with Ansible, though they'll receive slightly different ticket handling.

To locate where a module lives between 'core' and 'extras'
- Find the module at http://docs.ansible.com/list_of_all_modules.html
- Open the documentation page for that module
- If the bottom of the docs say "This is an extras module", submit your ticket to https://github.com/ansible/ansible-modules-extras
- Otherwise, submit your pull request to update the existing module to https://github.com/ansible/ansible-modules-core
- Note that python modules in ansible now also end in ".py" and this extension is required for new contributions.
- action_plugins (modules with server side components) still live in the main repo.  If your pull request touches both, which should be
   exceedingly rare, submit two new pull requests and make sure to mention the links to each other in the comments.

Otherwise, if this is a new module:
- Submit your pull request to add a module to https://github.com/ansible/ansible-modules-extras

It may be possible to re-patriate your pull requests automatically, one user-submitted approach for advanced git users
has been suggested at https://gist.github.com/willthames/afbaaab0c9681ed45619

Additionally, should you need more help with this, you can ask questions on:
- the development mailing list: https://groups.google.com/forum/#!forum/ansible-devel

Thanks you very much!
